assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +417,98 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per item. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
```

Comment:
That would be nice. But it's not a unique identifier or "primary key" so to speak.

 1. First of all several objects in kube don't even have a metadata.uid ... builtin services are an example here.
 2. Openshift objects share uids with their kubernetes counter parts. For example I created a project, and I get a Namespace with uid ```a6cb2a2c-0fa0-11e5-9c3e-10c37bdb8410``` and a Project with uid ```a6cb2a2c-0fa0-11e5-9c3e-10c37bdb8410``` ... ditto for ReplicationController vs Deployment.

Both of these can be seen as corner cases on their own but taken together they really limit the usefulness of ```metadata.uid```.